### PR TITLE
HIVE-1456: ClusterSync: Status & Message printer columns

### PR DESCRIPTION
**Pull request by 2uasimojo**
Make `oc get clustersync` print the overall status and message.

[HIVE-1456](https://issues.redhat.com/browse/HIVE-1456)

### CODE REVIEW

**Comment by 2uasimojo** [comment]
/hold

I haven't figured out how to test this yet.

File: apis/hiveinternal/v1alpha1/clustersync_types.go [hunk 1]
```diff
@@ -12,6 +12,8 @@ import (
 // +k8s:openapi-gen=true
 // +kubebuilder:subresource:status
 // +kubebuilder:resource:path=clustersyncs,shortName=csync,scope=Namespaced
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.conditions[0].reason`
+// +kubebuilder:printcolumn:name="Message",type=string,JSONPath=`.status.conditions[0].message`
```
Comment (2uasimojo) [inline apis/hiveinternal/v1alpha1/clustersync_types.go:16]: I assume these are just blank if there's no `.status.conditions[0]`. But afaict there always is.

**Comment by dgoodwin** [comment]
/assign @abutcher 

Nervous about Message in the columns, seems a bit too verbose, reason is ok though.

**Comment by 2uasimojo** [comment]
> Nervous about Message in the columns, seems a bit too verbose, reason is ok though.

Yeah; ideally I would like Message to be limited to `-o wide` -- but I need to figure out how to do that with kubebuilder.

[Later] Aha, `priority=1` ([ref](https://www.brendanp.com/posts/pretty-printing-with-kubebuilder/)]

**Comment by dgoodwin** [comment]
Ok that's handy.

**Comment by dgoodwin** [comment]
@2uasimojo do you want to plug that in here or would you like us to take over the PR.

**Comment by dgoodwin** [comment]
Ok no worries, that is fine with me.

**Comment by 2uasimojo** [comment]
Got an iteration of this working:

```
[efried@efried hive]$ oc get clustersync
NAME           STATUS
efried-leaf1   Failure
[efried@efried hive]$ oc get clustersync -o wide
NAME           STATUS    MESSAGE
efried-leaf1   Failure   SyncSet mygroup is failing
```

**Pull request merged**